File: benchmark/cases/go-sql-injection/source.go
// Case: SQL injection via fmt.Sprintf in a Go database query.
package store

import (
	"database/sql"
	"fmt"
)

func FindByEmail(db *sql.DB, email string) (*sql.Row, error) {
	// BUG: email is interpolated into the query string with fmt.Sprintf instead
	// of using parameterized placeholders, allowing SQL injection.
	q := fmt.Sprintf("SELECT id, email FROM users WHERE email = '%s'", email)
	row := db.QueryRow(q)
	return row, row.Err()
}

File: benchmark/cases/ts-sql-injection/source.ts
// Case: SQL injection via string concatenation in a TypeScript query builder.
import { db } from './db';

interface User {
  id: number;
  email: string;
}

export async function findUserByEmail(emailInput: string): Promise<User | null> {
  // BUG: user-controlled emailInput is concatenated directly into the SQL
  // string, allowing an attacker to break out of the quoted value and append
  // arbitrary SQL (e.g. "' OR '1'='1").
  const sql = `SELECT id, email FROM users WHERE email = '${emailInput}' LIMIT 1`;
  const rows = await db.query<User>(sql);
  return rows[0] ?? null;
}
